File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.java
import java.util.*;

class Solution {
    public int solution(String s) {
        
        String x = "";
        int answer = 0; //나눠지는 문자열의 수
        int same_count = 0; //x와 같은 문자의 수
        int diff_count = 0; //x와 다른 문자의 수
        
        boolean a = true;
        
        for(int i= 0; i<s.length(); i++){
            if(a) x = String.valueOf(s.charAt(i));
            
            if(x.equals(String.valueOf(s.charAt(i)))){
                same_count ++;
                a = false;
            }else{
                diff_count ++;
            }
            
            if(same_count ==diff_count){
                answer ++;
                same_count = 0;
                diff_count = 0;
                a = true; //x의 값을 바꿔주기 위해
            }
        }
        
        return (same_count == 0 && diff_count == 0) ? answer : answer + 1 ;
    }
}

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.js
function solution(s) {
    
    let same_count = 0; //x와 같은 문자 개수
    let different_count = 0; //x와 다른 문자 개수
    let answer = 0;
    
    let x = '';
    let check = true;
    
    for(let a of s.split("")){
        
        if(check){
            x = a;
            check = false;
        }
        
        if(x === a){
            same_count++;
        }else{
            different_count++;
        }
        
        if(same_count === different_count){ //개수가 같을 때
            check = true;
            same_count = 0;
            different_count = 0;
            answer++;
        }
    }
    
    return (same_count === 0 && different_count ===0)? answer: answer+1;
}